xlml/apis/task.py

```diff
@@ -155,23 +171,25 @@ def provision(self) -> Tuple[DAGNode, airflow.XComArg, airflow.XComArg]:
             self.task_test_config.benchmark_id, self.tpu_name_env_var
         )
         ssh_keys = ssh.generate_ssh_keys()
-
+      
+      output_location = name_format.generate_gcs_file_location(
+          self.task_test_config.benchmark_id
+      )
       queued_resource_op, queued_resource_name = tpu.create_queued_resource(
           tpu_name,
           self.task_gcp_config,
           ssh_keys,
           self.tpu_create_timeout,
           self.task_test_config,
       )
-      queued_resource_op >> tpu.ssh_tpu.override(task_id="setup")(
+      output_location >> queued_resource_op >> tpu.ssh_tpu.override(task_id="setup")(
```

Comment:
Will be a better logic if we put queued_resource_op >> metric_output_location >> setup? similar to GPU if apply

Reply:
The control dependency is added when the `output_location` is passed to `run_model`.